front-end/src/pages/TimelineSetupPage.js

```diff
@@ -35,69 +36,85 @@ const TimelineSetupPage = ({ onDataProcessed }) => {
   const [degrees, setDegrees] = useState([]);
   const navigate = useNavigate();
 
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const toggleModal = () => setIsModalOpen(!isModalOpen);
+
   useEffect(() => {
     if (isFirstRender.current) {
       onDataProcessed(); // Clear old timeline data on load
       isFirstRender.current = false;
     }
   }, [onDataProcessed]);
 
+  // get a list of all degrees by name
+  const getDegrees = async () => {
+    // TODO: Add proper error handling and user feedback for API failures
+    try {
+      const response = await api.post(`/degree/getAllDegrees`); //this should be a get request but backend is currently set to post
+      setDegrees(response.degrees);
+    } catch (err) {
+      Sentry.captureException(err);
+      console.error(err.message);
+    }
+  };
+
   useEffect(() => {
-    // get a list of all degrees by name
-    // TODO: Add loader while fetching degrees from API
-    const getDegrees = async () => {
-      // TODO: Add proper error handling and user feedback for API failures
-      try {
-        const response = await fetch(`${REACT_APP_SERVER}/degree/getAllDegrees`, {
-          method: 'POST',
-          headers: {
-            'Content-Type': 'application/json',
-          },
-        });
-
-        const jsonData = await response.json();
-        console.log('Degrees:', jsonData);
-        setDegrees(jsonData.degrees);
-      } catch (err) {
-        Sentry.captureException(err);
-        console.error(err.message);
-      }
-    };
     getDegrees();
   }, []);
 
-  const processFile = (file) => {
-    parsePdfFile(file).then((data) => {
-      const extractedData = extractAcceptanceDetails(data);
-      const transcriptData = extractedData.results;
-      const degree = extractedData.details?.degreeConcentration.toLowerCase() || 'Unknown Degree';
+  const processFile = async (file) => {
+    localStorage.setItem('Timeline_Name', null);
+    const formData = new FormData();
+    formData.append('file', file);
+    try {
+      const response = await api.upload(`/upload/parse`, formData);
```

Comment:
you don't need upload helper
```suggestion
      const response = await api.post('/upload/parse', formData, {
        headers: {
          'Content-Type': 'multipart/form-data',
        },
      });
```

Reply:
the content type will be overwritten, by what you parsed when calling the function